### PR TITLE
Fix Preview Theme Chat In Settings #1929

### LINKED ISSUE #1929: Fix Preview Theme Chat In Settings
**Actual Behaviour**

- Currently, Preview Theme Chat header(SUSI logo) out of position.
-Messaging a number of messages increases the height of the chat window.
- Max message width
- No, Send the message on 'enter' key press
- Allows to send an empty message
- Chat window width increase as the message increases, fix it

**Expected Behaviour**

Fix above bugs

**Screenshot**

screenshot 2019-01-05 at 4 47 36 am

screenshot 2019-01-05 at 4 49 14 am

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #1929 

Changes:

1. Added loading for message loading from SUSI side when API response is hit
2. Add Send the message on 'enter' key.
3. Check for empty messages(disallow)
4. Fix chat width container height.
5. Fix logo position

Demo Link: https://pr-1938-1-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by fragm3** [comment]
@akshatnitd @anshumanv please review

File: src/components/ChatApp/Settings/PreviewThemeChat.js [hunk 1]
```diff
@@ -3,43 +3,81 @@ import PropTypes from 'prop-types';
...
+      messages: [{ message: 'Hi, I am SUSI', author: 'SUSI', loading: false }],
       message: '',
     };
   }
 
   sendMessage = event => {
-    event.preventDefault();
-    this.addMessage(this.state.message, 'You');
+    if (this.state.message.trim().length > 0) {
+      this.addMessage(this.state.message, 'You');
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/PreviewThemeChat.js]: Please destructure state variables.

File: src/components/ChatApp/Settings/PreviewThemeChat.js [hunk 2]
```diff
@@ -3,43 +3,81 @@ import PropTypes from 'prop-types';
...
-            'SUSI',
-          );
-        }
-      })
-      .catch(error => {
-        console.log('Could not fetch reply');
-      });
-    this.setState({ message: '' });
+      let message = encodeURIComponent(this.state.message);
+      let self = this;
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/PreviewThemeChat.js]: Why is assigning this to self needed?

File: src/components/ChatApp/Settings/PreviewThemeChat.js [hunk 3]
```diff
@@ -3,43 +3,81 @@ import PropTypes from 'prop-types';
...
+      getSusiPreviewReply(message)
+        .then(payload => {
+          debugger;
+          let index;
+          for (var i = 0; i < self.state.messages.length; i++) {
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/PreviewThemeChat.js]: Use let

File: src/components/ChatApp/Settings/PreviewThemeChat.js [hunk 4]
```diff
@@ -3,43 +3,81 @@ import PropTypes from 'prop-types';
...
+              loading: false,
+            };
+          } else {
+            message = {
+              message: 'Sorry, I could not understand what you just said.',
+              author: 'SUSI',
+              loading: false,
+            };
+          }
+          self.setState(prevState => ({
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/PreviewThemeChat.js]: Use `this`

**Comment by fragm3** [comment]
@akshatnitd made the changes, for destructuring, it is not useful here. I tried destructuring,
screenshot 2019-01-13 at 2 34 22 pm

screenshot 2019-01-13 at 2 34 48 pm


See the messages, is not being read inside promise, due to scoping. So without destruring it is actually more clearer

File: src/components/ChatApp/Settings/PreviewThemeChat.js [hunk 5]
```diff
@@ -3,43 +3,79 @@ import PropTypes from 'prop-types';
...
-      .catch(error => {
-        console.log('Could not fetch reply');
-      });
-    this.setState({ message: '' });
+      let message = encodeURIComponent(this.state.message);
+      getSusiPreviewReply(message)
+        .then(payload => {
+          let index;
+          for (let i = 0; i < this.state.messages.length; i++) {
+            if (this.state.messages[i].loading === true) {
```
Comment (samagragupta) [inline src/components/ChatApp/Settings/PreviewThemeChat.js]: Please remove `=== true`

File: src/components/ChatApp/Settings/PreviewThemeChat.js [hunk 6]
```diff
@@ -3,43 +3,79 @@ import PropTypes from 'prop-types';
...
-            'Sorry, I could not understand what you just said.',
-            'SUSI',
-          );
-        }
-      })
-      .catch(error => {
-        console.log('Could not fetch reply');
-      });
-    this.setState({ message: '' });
+      let message = encodeURIComponent(this.state.message);
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/PreviewThemeChat.js]: Rename variable to some other name, and I see destructuring clearly working out.

**Comment by fragm3** [comment]
@akshatnitd it's not working.
See my code:

screenshot 2019-01-16 at 2 59 01 am

Try sending a message at the deployed link: https://www.skillful-impulse.surge.sh

**Comment by fragm3** [comment]
@akshatnitd on destructuring inside promise as well, I am getting this eslint error(code working fine):
screenshot 2019-01-18 at 2 00 17 pm

File: src/components/ChatApp/Settings/PreviewThemeChat.js [hunk 7]
```diff
@@ -3,43 +3,81 @@ import PropTypes from 'prop-types';
...
-      messages: [{ message: 'Hi I am Susi', author: 'Susi' }],
+      messages: [{ message: 'Hi, I am SUSI', author: 'SUSI', loading: false }],
       message: '',
     };
   }
 
   sendMessage = event => {
-    event.preventDefault();
-    this.addMessage(this.state.message, 'You');
+    const {message, messages} = this.state
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/PreviewThemeChat.js]: Remove `messages` from here. The eslint error is due to this.

**Comment by fragm3** [comment]
@akshatnitd made the changes

**Comment by fragm3** [comment]
@samagragupta I tried resizing it, it is looking perfectly fine to me, unable to reproduce your error

**Pull request merged**